src/python/pants/backend/native/tasks/native_external_library_fetch.py

```diff
@@ -0,0 +1,267 @@
+    match_group = re.match( r"^lib(.*)\.(a|so|dylib)$", filename)
+    if match_group:
+      return match_group.group(1)
+    return None
+
+  @classmethod
+  def register_options(cls, register):
+    super(NativeExternalLibraryFetch, cls).register_options(register)
+    register('--conan-remotes', type=list, default=['https://conan.bintray.com'], advanced=True,
+             fingerprint=True, help='The conan remote to download conan packages from.')
+
+  @classmethod
+  def subsystem_dependencies(cls):
+    return super(NativeExternalLibraryFetch, cls).subsystem_dependencies() + (Conan,)
+
+  @classmethod
+  def product_types(cls):
+    return [cls.NativeExternalLibraryFiles]
+
+  @property
+  def cache_target_dirs(self):
+    return True
+
+  @memoized_property
+  def _conan_binary(self):
+    return Conan.global_instance().bootstrap_conan()
+
+  def execute(self):
+    task_product = self.context.products.get_data(self.NativeExternalLibraryFiles,
+                                                  self.NativeExternalLibraryFiles)
+
+    native_lib_tgts = self.context.targets(self.native_library_constraint.satisfied_by)
+    if native_lib_tgts:
+      with self.invalidated(native_lib_tgts,
+                            invalidate_dependents=True) as invalidation_check:
+        resolve_vts = VersionedTargetSet.from_versioned_targets(invalidation_check.all_vts)
+        vts_results_dir = self._prepare_vts_results_dir(resolve_vts)
+        if invalidation_check.invalid_vts or not resolve_vts.valid:
+          for vt in invalidation_check.all_vts:
+            self._fetch_packages(vt, vts_results_dir)
+        self._populate_task_product(vts_results_dir, task_product)
+
+  def _prepare_vts_results_dir(self, vts):
+    """
+    Given a `VergetTargetSet`, prepare its results dir.
+    """
+    vt_set_results_dir = os.path.join(self.workdir, vts.cache_key.hash)
+    safe_mkdir(vt_set_results_dir)
+    return vt_set_results_dir
+
+  def _populate_task_product(self, results_dir, task_product):
+    """
+    Sets the relevant properties of the task product (`NativeExternalLibraryFiles`) object.
+    """
+    lib = os.path.join(results_dir, 'lib')
+    include = os.path.join(results_dir, 'include')
+
+    if os.path.exists(lib):
+      task_product.lib = lib
+      for filename in os.listdir(lib):
+        lib_name = self._parse_lib_name_from_library_filename(filename)
+        if lib_name:
+          task_product.add_lib_name(lib_name)
+
+    if os.path.exists(include):
+      task_product.include = include
+
+  def ensure_conan_remote_configuration(self, conan_binary):
+    """
+    Ensure that the conan registry.txt file is sanitized and loaded with
+    a pants-specific remote for package fetching.
+
+    :param conan_binary: The conan client pex to use for manipulating registry.txt.
+    """
+
+    # Delete the conan-center remote from conan's registry.
+    remove_conan_center_remote_cmdline = conan_binary.pex.cmdline(['remote',
```

Comment:
This is because Conan will prepend this when bootstrapped for the first time. We do this to remove the conan-center remote and use the Pants-default (which is conan-center, but we control it).

Reply:
I feel like adding a comment like "Conan will prepend this when bootstrapped for the first time" here would make this more clear.